frontend/src/app/output/navigate/navigate-page.component.ts

```diff
@@ -60,6 +58,16 @@ export class NavigatePageComponent implements OnInit, AfterViewChecked {
 
   ngOnInit() {
     this.initialPath = this.route.snapshot.paramMap.get('path');
+    this.route.paramMap.subscribe((params: ParamMap) => {
+      this.currentRoute = params.get('name');
+      this.hierarchyService.hierarchy().subscribe(
```

Comment:
It is a  bad practice to do a subscribe inside a subscribe. You should flatten this. Read 21.2. Leverage operators from NinjaSquad book.